File: 位运算/1720.py
class Solution:
    def decode(self, encoded, first: int):
        res = [first]
        pre = first
        for num in encoded:
            ans = pre^num
            res.append(ans)
            pre = ans
        return res

encoded = [1,2,3]
first = 1
obj = Solution()
print(obj.decode(encoded,first))

File: 位运算/1720.cpp
#include<vector>
using namespace std;
class Solution {
public:
    vector<int> decode(vector<int>& encoded, int first) {
        vector<int>res;
        res.push_back(first);
        for(int i=0;i<encoded.size();i++)
            res.push_back(res.back()^encoded[i]);
        return res;
    }
};